server/app/services/applicant/ApplicantService.java

```diff
@@ -428,33 +429,69 @@ private CompletionStage<ReadOnlyApplicantProgramService> stageAndUpdateIfValid(
    *     ApplicationSubmissionException} is thrown and wrapped in a `CompletionException`.
    */
   public CompletionStage<ApplicationModel> submitApplication(
-      long applicantId, long programId, CiviFormProfile submitterProfile, Request request) {
+      long applicantId, long programId, CiviFormProfile submitterProfile, Request request)
+      throws ProgramNotFoundException {
+    ProgramDefinition pd = programService.getFullProgramDefinition(programId);
```

Comment:
suggestion (non-blocking): You could try/catch the `ProgramNotFoundException` in this method to avoid having to bubble it up and make changes to the `ApplicantProgramReviewController` but I don't feel strongly about it.